pkg/privatekey/config.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2025 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package privatekey
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+// Config defines the set of environment variables required
+// for running the private key job.
+type Config struct {
+	ProjectID      string
+	Location       string
+	KeyRing        string
+	Key            string
+	PrivateKeyFile string
+}
+
+// Validate validates the config after load.
+func (cfg *Config) Validate() (vErr error) {
+	if cfg.ProjectID == "" {
+		vErr = errors.Join(vErr, fmt.Errorf("PROJECT_ID is required"))
+	}
+
+	if cfg.Location == "" {
+		vErr = errors.Join(vErr, fmt.Errorf("LOCATION is required"))
+	}
+
+	if cfg.KeyRing == "" {
+		vErr = errors.Join(vErr, fmt.Errorf("KEY_RING is required"))
+	}
+
+	if cfg.Key == "" {
+		vErr = errors.Join(vErr, fmt.Errorf("KEY is required"))
+	}
+
+	if cfg.PrivateKeyFile == "" {
+		vErr = errors.Join(vErr, fmt.Errorf("PRIVATE_KEY_FILE is required"))
+	}
+	return
+}
+
+// ToFlags binds the config to the [cli.FlagSet] and returns it.
+func (cfg *Config) ToFlags(set *cli.FlagSet) *cli.FlagSet {
+	f := set.NewSection("COMMON JOB OPTIONS")
+
+	f.StringVar(&cli.StringVar{
+		Name:   "project-id",
+		Target: &cfg.ProjectID,
+		EnvVar: "PROJECT_ID",
+		Usage:  `The gcp project id.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "location",
+		Target: &cfg.Location,
+		EnvVar: "LOCATION",
+		Usage:  `The Cloud KMS location of the key ring.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "key-ring",
+		Target: &cfg.KeyRing,
+		EnvVar: "KEY_RING",
+		Usage:  `the name of the key ring that contains the key.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "key",
+		Target: &cfg.Key,
+		EnvVar: "KEY",
+		Usage:  `the name of the key.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "private-key-file",
+		Target: &cfg.PrivateKeyFile,
+		EnvVar: "PRIVATE_KEY_FILE",
```

Comment:
The problem with requiring a filepath is that it requires any automation to write the key to disk before running the command. I think we need the ability to pass the key directly.

Reply:
Probably there is some gap about the CUJ. 
The CUJ we plan is that 
1. users download the github private key to local disk
2. users use the minty cli command to import the private the key to cloud kms

We don't plan the automation to write the key to disk. Users just click github UI to download the private key. The ability to pass the key directly is blocked now as there is no official github api to do it.

Reply:
I think we can support both. The patterns I usually use are:

- Make the variable accept a string, and if the string starts with `@`, assume it's a filepath (and read contents from disk)

- Make the variable accept a filepath, and if input is exactly `-`, read the value from stdin instead

Reply:
Got it. I will use the approach to make the variable accept a filepath, and if input is exactly -, read the value from stdin instead as I think filepath is more common here as when users download the github private key, by default it is a file.

Reply:
I've changed the name to indicate it support both. Will implement it in later PR.